src/LayerEditor/ui/panels/regions_panel.py

```diff
@@ -463,12 +446,18 @@ def _update_region_content(self):
 
 
     # ======= Internal signal handlers
-    def _combo_set_region(self, region_id):
+    def _combo_set_region(self, combo_id):
         """
         Handle change in region combo box.
         """
-        self._region_id = region_id
-        self.layer_heads.select_regions({self.layer_id: region_id})
+        for reg_id, idx in self._combo_id_to_idx.items():
+            if idx == combo_id:
+                break
+        region = self.layer_heads.regions_model.regions.get(reg_id)
+        self.layer.gui_selected_region = region
+        self.layer.set_region_to_selected_shapes(region)
```

Comment:
Layer can have just one method to select a region.

Reply:
I don't want this change to register in undo redo system,  if that is the issue.. So there are two ways to select region.